### PR TITLE
test(solve): add docstrings and tests for solve module

### PR DESCRIPTION
**Pull request by lsetiawan**
## Overview

This PR add additional docstring, typehints, and tests for the `solve.py` module.

### CODE REVIEW

**Comment by lsetiawan** [comment]
This relates to https://github.com/uw-ssec/offshore-geodesy/issues/50

File: src/gnatss/ops/solve.py [hunk 1]
```diff
@@ -287,12 +409,35 @@ def solve_transponder_locations(
...
+        the GPS covariance matrix, and the observed delays for a single ping.
+    transponders_mean_sv : (N,) ndarray
+        The mean signal velocity for each transponder.
+    transponders_xyz : (N,3) ndarray
+        The XYZ positions of each transponder.
+    transponders_delay : (N,) ndarray
+        The internal delay for each transponder.
+    travel_times_variance : float
+        The variance in travel times
+
```
Comment (carlosgjs) [inline src/gnatss/ops/solve.py:434]: nit: either end all or none with a period (for a single sentence usually no period is fine)

File: tests/ops/test_solve.py [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+@settings(deadline=None)
+def test__calc_unit_vectors(vectors) -> None:
+    """Test calculate unit vectors"""
+    result_array = _calc_unit_vectors(vectors)
+
+    assert result_array.shape == vectors.shape
+
+
+@given(
+    # Currently hard-coded to 3 transponders
```
Comment (carlosgjs) [inline tests/ops/test_solve.py:90]: Is this a TODO?

Comment (carlosgjs) [reply tests/ops/test_solve.py:90]: Maybe add a top level constant N_TRANSPONDERS=3

**Review by carlosgjs — APPROVED** [review]
just some minor comments

**Pull request merged**